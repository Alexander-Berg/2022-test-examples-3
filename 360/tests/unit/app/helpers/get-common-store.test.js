import getCommonStore from '../../../../app/helpers/get-common-store';

jest.mock('@ps-int/ufo-server-side-commons/helpers/bindings-node14');

describe('app/helpers/get-common-store =>', () => {
    const mockedReq = {
        protocol: 'https:',
        get: (param) => param === 'host' ? 'disk.yandex.ru' : undefined,
        originalUrl: '/notes',
        tld: 'ru',
        fullTld: 'ru',
        cookies: {
            yandexuid: '56789000'
        },
        experiments: {
            metrika: 'exp-metrika',
            diskFlags: {
                disk_notes_prod_experiment: true
            }
        },

        ua: {
            BrowserName: 'YandexBrowser',
            OSFamily: 'Ubuntu'
        },
        user: {
            id: '12345',
            auth: true,
            name: '%username%'
        },
        yandexServices: {
            disk: 'https://disk.yandex.ru',
            passport: 'https://passport.yandex.ru'
        },
        lang: 'ru',
        generatedSK: 'generated-sk',
        generatedNoSaltSK: 'generated-no-salt-sk',
        nonce: 'nonce-123'
    };

    it('should map request fields to store', () => {
        expect(getCommonStore(mockedReq)).toMatchSnapshot();
    });

    it('should set errorCode', () => {
        const store = getCommonStore(mockedReq, { errorCode: 500 });
        expect(store.errorCode).toEqual(500);
    });

    it('should fallback if req.experiments does not exist', () => {
        const reqWithoutExperiments = Object.assign(mockedReq);
        delete reqWithoutExperiments.experiments;
        const store = getCommonStore(reqWithoutExperiments);
        expect(store.environment.experiments).toEqual({
            metrika: '',
            flags: {}
        });
    });

    it('should set balloonMode', () => {
        const store = getCommonStore(mockedReq, { balloonMode: true });
        expect(store.environment.balloonMode).toBe(true);
    });

    it('should set broMode', () => {
        const store = getCommonStore(mockedReq, { broMode: true });
        expect(store.environment.broMode).toBe(true);
    });
});
