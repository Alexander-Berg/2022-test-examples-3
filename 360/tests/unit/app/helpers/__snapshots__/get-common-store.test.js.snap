// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/helpers/get-common-store => should map request fields to store 1`] = `
Object {
  "environment": Object {
    "availableLangs": Array [
      "ru",
      "en",
      "uk",
      "tr",
    ],
    "avatarsOrigin": "https://avatars.mdst.yandex.net",
    "balloonMode": false,
    "broMode": false,
    "currentLang": "ru",
    "currentUrl": "https:://disk.yandex.ru/notes",
    "experiments": Object {
      "flags": Object {
        "disk_notes_prod_experiment": true,
      },
      "metrika": "exp-metrika",
    },
    "fullTld": "ru",
    "isWebViewMail360": false,
    "noSaltSk": "generated-no-salt-sk",
    "nonce": "nonce-123",
    "scripts": Object {
      "datasync": undefined,
    },
    "sk": "generated-sk",
    "tld": "ru",
    "usePSHeader": true,
    "windowWidth": undefined,
    "yandexuid": "56789000",
  },
  "errorCode": null,
  "services": Object {
    "disk": "https://disk.yandex.ru",
    "passport": "https://passport.yandex.ru",
  },
  "settings": Object {
    "showDocsInPsHeader": false,
  },
  "ua": Object {
    "BrowserName": "YandexBrowser",
    "OSFamily": "Ubuntu",
  },
  "user": Object {
    "auth": true,
    "id": "12345",
    "name": "%username%",
    "settings": Object {},
  },
}
`;
